src/inc/dacprivate.h

```diff
@@ -97,14 +60,14 @@ struct MSLAYOUT DacpObjectData : ZeroInit<DacpObjectData>
 
     CLRDATA_ADDRESS RCW;
     CLRDATA_ADDRESS CCW;
-    
+
     HRESULT Request(ISOSDacInterface *sos, CLRDATA_ADDRESS addr)
     {
         return sos->GetObjectData(addr, this);
     }
 };
 
-struct MSLAYOUT DacpExceptionObjectData : ZeroInit<DacpExceptionObjectData>
```

Comment:
Is there some rationale I should apply about why removing the ZeroInit is safe? Superficially it appears that previously we were zero initting our data, now we are not zero initting, and maybe this has caused some bugs because the consuming code required the zero init for correctness?

My hunch is that it would be easier to define a default constructor for each type which does the zero init rather than prove via code review that the zero init wasn't required for correctness to begin with.

Reply:
Even though this h file has "private" in its name it actually is the internal interface between SOS and the DAC.  SOS now resides in a different [repo](https://github.com/dotnet/diagnostics.git) so the alignment that MSLAYOUT does (or did) needs to be the same before/after this change.

I think the DAC API assumes that this structs are zero initialized when passed by SOS.

Reply:
I have initialized the fields of each struct:

* BOOL -> FALSE
* integers -> 0
* enums -> 0-index value
* arrays -> {}

during my testing, in one case, `static_assert` on top of this file was violated, which helped me validate the change in terms of: struct sizes are same before/after.